pyteamcity/future/artifact.py

```diff
@@ -0,0 +1,93 @@
+import fnmatch
+import os
+
+from .core.utils import parse_date_string
+
+
+class Artifact(object):
+    def __init__(self, build, path=''):
+        self.build = build
+        self.path = path
+        teamcity = self.build.build_query_set.teamcity
+        url = self.build.api_url + '/artifacts/metadata/' + self.path
+        res = teamcity.session.get(url)
+        res.raise_for_status()
+        self._data = res.json()
+
+    @property
+    def name(self):
+        return self._data['name']
+
+    def getsize(self):
+        return self._data.get('size')
+
+    @property
+    def size(self):
+        return self.getsize()
+
+    @property
+    def modification_time(self):
+        return parse_date_string(self._data['modificationTime'])
+
+    def splitext(self):
+        return os.path.splitext(self.name)
+
+    @property
+    def ext(self):
+        return self.splitext()[1]
+
+    def fnmatch(self, pattern):
+        return fnmatch.fnmatch(self.name, pattern)
```

Comment:
I don't have a use case in mind for `Artifact.fnmatch()`. I implemented it to have parity with path.py.

The `ArtifactFinder` is a good idea, though I think there are some methods here that do that (again, I borrowed them from path.py). Look at the `dirs`, `files`, and `listdir` methods.